core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -679,12 +680,25 @@ private[spark] object Utils extends Logging {
    * If no directories could be created, this will return an empty list.
    */
   private[spark] def getOrCreateLocalRootDirs(conf: SparkConf): Array[String] = {
+    if (localRootDirs == null) {
```

Comment:
(On a side note, this patch might do some weird things if the method is called with different confs.)